### PR TITLE
add allExcept shorthand methods in ReqSpacesInAnon

### LINKED ISSUE #1513: requireSpacesInAnonymousFunctionExpression: Add exception for shorthand methods
See discussion in https://github.com/jscs-dev/node-jscs/issues/1470#issuecomment-118666767

Need to support `allExcept: ['shorthandMethods']`

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/23862586-requirespacesinanonymousfunctionexpression-add-exception-for-shorthand-methods?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ksheedlo**
Fixes #1513.

### CODE REVIEW

File: lib/rules/require-spaces-in-anonymous-function-expression.js [hunk 1]
```diff
@@ -3,8 +3,9 @@
  *
  * Type: `Object`
  *
- * Values: `"beforeOpeningRoundBrace"` and `"beforeOpeningCurlyBrace"` as child properties.
- * Child properties must be set to `true`.
+ * Values: `"beforeOpeningRoundBrace"` and `"beforeOpeningCurlyBrace"` as child
```
Comment (markelog) [inline lib/rules/require-spaces-in-anonymous-function-expression.js]: We started to clean-up docs a little, could you follow the same style as in - https://github.com/jscs-dev/node-jscs/pull/1667/files

?

Comment (ksheedlo) [inline lib/rules/require-spaces-in-anonymous-function-expression.js]: sure!

File: lib/rules/require-spaces-in-anonymous-function-expression.js [hunk 2]
```diff
@@ -117,7 +153,7 @@ module.exports.prototype = {
                     message: 'Missing space before opening curly brace'
                 });
             }
-        });
+        }.bind(this));
```
Comment (markelog) [inline lib/rules/require-spaces-in-anonymous-function-expression.js]: Nit-pick: we usually don't use bindings, that is why we have those lines:
```js
var beforeOpeningRoundBrace = this._beforeOpeningRoundBrace;
var beforeOpeningCurlyBrace = this._beforeOpeningCurlyBrace;
```
Would you mind using `this.exceptions` property in the similar manner?

File: lib/rules/require-spaces-in-anonymous-function-expression.js [hunk 3]
```diff
@@ -42,6 +43,10 @@ module.exports = function() {};
 
 module.exports.prototype = {
     configure: function(options) {
+        this.exceptions = {
```
Comment (markelog) [inline lib/rules/require-spaces-in-anonymous-function-expression.js]: We pretty strict about naming conventions of the private properties, could name it `._exceptions`?

**Comment by markelog** [comment]
Minor notes, looks good otherwise

**Comment by ksheedlo** [comment]
Update: Responded to review comments.

**Comment by ksheedlo** [comment]
Update: Had to add a coverage hack for the `allExcept: true` case.

**Comment by markelog** [comment]
Change the name of the exception, anyway, good work, thank you

**Comment by hzoo** [comment]
Looks like we forgot to do the same for the disallow rule? @ksheedlo If you don't have time to do it I can

**Comment by ksheedlo** [comment]
I'm swamped lately with work and other projects, so I'd appreciate if
someone else took care of it.

**Pull request closed without merging**